app/src/contexts/Context.tsx

```diff
@@ -1,16 +1,20 @@
-import AssetTypeProvider from "./Asset/AssetTypeContext";
+import AssetTypeProvider from "./asset/AssetTypeContext";
+import ProjectProvider from "./project/ProjectContext";
+import ProjectDataProvider from "./project/ProjectDataContext";
 
 /**
  * This is used to wrap all contexts into one component.
  * @param props Allows nested components
  * @returns <Context> </Context>
  */
 export default function Contexts(props: any) {
-    return (
-        <>
-            <AssetTypeProvider>
-                {props.children}
-            </AssetTypeProvider>
-        </>
-    )
-}
\ No newline at end of file
+  return (
+    <>
+      <AssetTypeProvider>
+        <ProjectProvider>
+          <ProjectDataProvider>{props.children}</ProjectDataProvider>
```

Comment:
This is going to get quite long/deep lol, but don't do anything about it on this PR, I think I have an idea I might play with.